### PR TITLE
expose partner config to the sdk

### PR DESCRIPTION
**Pull request by OnyekaIjeh**

### CODE REVIEW

File: test/rskregistrar/partnerConfiguration.json [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+}
```
Comment (sergio-jara) [inline test/rskregistrar/partnerConfiguration.json:406]: we agreed on not using this 😬

File: src/PartnerConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  getPrice (name: string, expires: BigNumber, duration: BigNumber): Promise<BigNumber> {
+    return this.partnerConfiguration.getPrice(name, expires, duration)
+  }
+
+  validateName (name: string, duration: BigNumber): Promise<void> {
+    return this.partnerConfiguration.validateName(name, duration)
+  }
+
+  async setDiscount (discount: BigNumber, signer?: Signer): Promise<void> {
```
Comment (sergio-jara) [inline src/PartnerConfiguration.ts:124]: even tho the current implementation doesnt have it, can we use the same documentation style we use in the Solidity code? aiming to automate the docs generation and intellisense when using the functions.

File: src/PartnerConfiguration.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  getSigner (signer?: Signer): Signer {
+    const _signer = signer || this.signer
+
+    if (!_signer) {
+      throw new Error('Signer is not defined')
+    }
+
+    return _signer
```
Comment (sergio-jara) [inline src/PartnerConfiguration.ts:218]: nice! at some point we might reuse this logic in other classes I think

File: src/PartnerConfiguration.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  'function setMaxLength(uint256 maxLength) external',
+  'function setMinDuration(uint256 minDuration) external',
+  'function setMinLength(uint256 minLength) external',
+  'function validateName(string memory name, uint256 duration) external view'
+]
+
+export class PartnerConfiguration {
+  private readonly partnerConfiguration: Contract
+
+  constructor (private readonly partnerConfigurationAddress: string, private readonly signer?: Signer, private readonly provider: string = 'http://localhost:8545') {
```
Comment (sergio-jara) [inline src/PartnerConfiguration.ts]: are we going to defauit to localhost? maybe testnet?

**Review by sergio-jara — CHANGES_REQUESTED** [review]

File: src/PartnerConfiguration.ts (hunk 4 above)
Comment (OnyekaIjeh) [reply src/PartnerConfiguration.ts]: Testnet

Comment (sergio-jara) [reply src/PartnerConfiguration.ts]: lol but I can still see the localhost address in the constructor xD

**Review by sergio-jara — APPROVED** [review]

File: src/PartnerConfiguration.ts (hunk 4 above)
Comment (OnyekaIjeh) [reply src/PartnerConfiguration.ts]: this is outdated

**Pull request merged**